client/pov-recovery/src/lib.rs

```diff
@@ -105,26 +132,84 @@ pub struct RecoveryDelay {
 	pub max: Duration,
 }
 
+impl RecoveryDelay {
```

Comment:
I know you didn't add this type, but it might be nice to rename it as it's a little confusing. Reading `RecoveryDelay` I thought that it contains a delay for the recovery as the name suggests, but it actually contains the range the delay can take. So maybe `RecoveryDelayRange`?